primefaces/src/main/java/org/primefaces/component/steps/StepsRenderer.java

```diff
@@ -150,7 +150,14 @@ protected void encodeMenuItem(FacesContext context, Steps steps, MenuItem menuit
 
         writer.startElement("span", steps);
         writer.writeAttribute("class", Steps.STEP_NUMBER_CLASS, null);
-        writer.writeText((index + 1), null);
+        if (menuitem.getIcon() != null) {
```

Comment:
It's better to use LangUtils.isEmpty here

Reply:
Right, that way empty icon="" attributes are also handled correctly.  thx